### PR TITLE
Support Toast notifications for async actions like `Delete` or `Restart`

### PR DESCRIPTION
**Pull request by lawrencegripper**
#15 

Bonus, fix some buggy code in the `eventing` package and update some docs. 

Todo: 
- [ ] ~~Continue refactoring of `eventing.SendStatusEvent`~~ For another PR
- [x] Improve test coverage (integration tests)
- [x] Check behavior on things like `restart` and `shutdown`

### CODE REVIEW

File: internal/pkg/views/statusbar.go [hunk 1]
```diff
@@ -124,11 +124,11 @@ func (w *StatusbarWidget) Layout(g *gocui.Gui) error {
 	}
 
 	if w.currentMessage.InProgress {
-		fmt.Fprint(v, style.Loading("⏳  "+w.currentMessage.Message))
+		fmt.Fprint(v, style.Loading(w.currentMessage.Icon()+"  "+w.currentMessage.Message))
```
Comment (jjcollinge) [inline internal/pkg/views/statusbar.go]: This condition has been moved into the `Icon()` method so arguably isn't required (atleast for the `InProgress` and `Failure`, might still want then to apply a different style to `Completed`)

Comment (lawrencegripper) [reply internal/pkg/views/statusbar.go]: Good spot - will fix up. Thanks!

Comment (lawrencegripper) [reply internal/pkg/views/statusbar.go]: Done - thanks!

**Comment by lawrencegripper** [comment]
I'm much happier with this now, only remaining work is to remove the panics and improve the test coverage.

**Comment by lawrencegripper** [comment]
I'm pretty happy this is in a fairly good working state. It's a pretty broad PR now so my preference would be to merge and revisit the remaining work in a separate PR

**Pull request merged**